modules/vault-secrets.nix

```diff
@@ -206,41 +201,47 @@ in
-          chmod 0755 "${cfg.outPathPrefix}"
+          # Ensure the base path exists and has the correct permissions
+          mkdir -p "${cfg.outPrefix}"
+          chmod 0755 "${cfg.outPrefix}"
 
           # Make sure we start from a clean slate
           rm -rf "${secretsPath}"
           mkdir -p "${secretsPath}"
 
+          # Log into Vault using credentials from environmentFile
           vaultOutput="$(vault write -format=json auth/approle/login role_id="$VAULT_ROLE_ID" secret_id=- <<< "$VAULT_SECRET_ID")"
           jq '.auth.client_token = "redacted"' <<< "$vaultOutput"
-
           VAULT_TOKEN="$(jq -r '.auth.client_token' <<< "$vaultOutput")"
           export VAULT_TOKEN
+
         '' + optionalString (secretsKey != null) ''
-          json_dump="$(vault kv get -format=json "${cfg.vaultPathPrefix}/${namespace}/${name}/${secretsKey}" || true)"
+          json_dump="$(vault kv get -format=json "${cfg.vaultPrefix}/${name}/${secretsKey}" || true)"
           if [[ -n "$json_dump" ]]; then
-        '' + (if secretsBase64 then ''
+        '' + (if secretsAreBase64 then ''
               dumpsecrets="$(jq -r 'select(.data.data != null) | .data.data | to_entries[] | "base64 -d <<< \"\(.value)\" > ${secretsPath}/\(.key)"' <<< "$json_dump")"
         '' else ''
               dumpsecrets="$(jq -r 'select(.data.data != null) | .data.data | to_entries[] | "builtin printf \"%s\\n\" \"\(.value)\" > ${secretsPath}/\(.key)"' <<< "$json_dump")"
         '') + optionalString (secretsKey != null) ''
-              echo "Found secrets at ${cfg.vaultPathPrefix}/${namespace}/${name}/secrets (''${#dumpsecrets} bytes)" >&2
+              echo "Found secrets at ${cfg.vaultPrefix}/${name}/secrets (''${#dumpsecrets} bytes)" >&2
               eval "$dumpsecrets"
           fi
+
         '' + optionalString (environmentKey != null) ''
-          json_dump="$(vault kv get -format=json "${cfg.vaultPathPrefix}/${namespace}/${name}/${environmentKey}" || true)"
+          json_dump="$(vault kv get -format=json "${cfg.vaultPrefix}/${name}/${environmentKey}" || true)"
           if [[ -n "$json_dump" ]]; then
-              jq -r '.data.data | to_entries[] | "${optionalString (environmentPrefix != null) "${toUpper environmentPrefix}_"}\(.key)=\"\(.value)\""' <<< "$json_dump" > "${secretsPath}/environment"
+              jq -r '.data.data | to_entries[] | "${optionalString (environmentVariableNamePrefix != null) "${toUpper environmentVariableNamePrefix}_"}\(.key)=\"\(.value)\""' <<< "$json_dump" > "${secretsPath}/environment"
               echo "Dumped environment file at ${secretsPath}/environment" >&2
           fi
-        '' + ''
+
+        '' + optionalString (extraScript != "") ''
           secretsPath="${secretsPath}"
+          source <<(jq -r '.data.data | to_entries[] | "${optionalString (environmentVariableNamePrefix != null) "export ${toUpper environmentVariableNamePrefix}_"}\(.key)=\"\(.value)\""' <<< "$json_dump")
```

Comment:
Looks like `export` is intended be outside `optionalString`

Reply:
It's intended to be inside. It exports the variables for use in `extraScript`, so it should only run if `extraScript` is defined.

Reply:
I mean that `export` keyword is used only when `environmentVariableNamePrefix` is not null, shouldn't it always be used?

Reply:
Oh, _that_ `optionalString`. Yep, good catch. Pushed a fixup.